kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -257,7 +261,10 @@ trait ProcBuilder {
       logCaptureThread = null
     }
     if (destroyProcess && process != null) {
-      process.destroyForcibly()
+      process.destroy()
+      if (!process.waitFor(engineStartupDestroyTimeout, TimeUnit.MICROSECONDS)) {
+        process.destroyForcibly()
```

Comment:
we'd better add a log here to warn that we are going to forcibly kill process `pid` due to the process does not exit after xxx s.

There is a tricky in getting pid from `Process` in Java8, but is easy since 9, we may need to introduce a method in `ProcessUtils`

Reply:
ok，I try to fix it